### PR TITLE
Fixed FDA old bug that stopped phasequad

### LINKED ISSUE #26270: Frequency Domain Analysis Old MaskedDetectors Error
Frequency Domain Analysis Old in MantidPlot raises an error when trying to transform PhaseQuad
### Steps to reproduce the behavior
1. Open Mantidplot
1. Open Muon Analysis (the originalone not muon analysis 2)
1. load MUSR 62260
1. Open frequency domain analysis old
1. Change workspace to PhaseQuad
1. Click Calculate FFT
1. Mantid raises an error "MaskedDetectors"
### Platforms affected
Windows

### PR DESCRIPTION
**Pull request by AnthonyLim23**
The phaseQuad option was not working in the old FDA GUI. 

**To test:**

- MUSR 62260
    - Open Mantidplot
    - Open Muon Analysis (the original one not muon analysis 2)
    - load MUSR 62260
    - Open frequency domain analysis old
    - Change workspace to PhaseQuad
    - Click Calculate FFT
    - [x] Mantid should not raises an error "MaskedDetectors"
    - [x] The PhaseQuad workspace should be in the MUSR62260 group
    - [x] Check that an FFT of the group also adds a workspace
    - Close the frequency GUI

- HIFI 84447
    - In muon analysis load HIFI 84447
    - Open frequency domain analysis old
    - Change workspace to PhaseQuad
    - Click Calculate FFT
    - [x] Mantid should not raises an error "MaskedDetectors"
    - [x] The PhaseQuad workspace should be in the HIFI84447 group
    - [x] Check that an FFT of the group also adds a workspace

Fixes #26270. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_model.py [hunk 1]
```diff
@@ -99,15 +99,12 @@ def preAlg(self, preInputs):
         """
         self.alg = mantid.AlgorithmManager.create("PaddingAndApodization")
         self.alg.initialize()
-        self.alg.setAlwaysStoreInADS(False)
         self.alg.setRethrows(True)
 
         for name, value in iteritems(preInputs):
+            print(name, value)
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_model.py]:
```suggestion
```

File: scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_model.py [hunk 2]
```diff
@@ -134,48 +127,48 @@ def makePhaseQuadTable(self, inputs):
         """
         generates a phase table from CalMuonDetectorPhases
         """
-        cloned_workspace_CalMuon = mantid.CloneWorkspace(InputWorkspace=inputs["InputWorkspace"], StoreInADS=False)
-        mantid.MaskDetectors(Workspace=cloned_workspace_CalMuon, DetectorList=inputs['MaskedDetectors'], StoreInADS=False)
+        cloned_workspace_CalMuon = mantid.CloneWorkspace(InputWorkspace=inputs["InputWorkspace"], OutputWorkspace="__tmp__")
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_model.py]:
```suggestion
        mantid.CloneWorkspace(InputWorkspace=inputs["InputWorkspace"], OutputWorkspace="__tmp__")
```

File: scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_model.py [hunk 3]
```diff
@@ -134,48 +127,48 @@ def makePhaseQuadTable(self, inputs):
...
+        mantid.DeleteWorkspace("__tmp__")
+        mantid.DeleteWorkspace("fits")
         self.alg = None
 
     def PhaseQuad(self, inputs):
         """
         do the phaseQuad algorithm
         groups data into a single set
         """
+        # falls over in here
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_model.py]:
```suggestion
```

File: scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_model.py [hunk 4]
```diff
@@ -134,48 +127,48 @@ def makePhaseQuadTable(self, inputs):
...
         cloned_workspace = mantid.CloneWorkspace(InputWorkspace=inputs["InputWorkspace"], StoreInADS=False)
         mantid.MaskDetectors(Workspace=cloned_workspace, DetectorList=inputs['MaskedDetectors'], StoreInADS=False)
         cropped_workspace_pre_phasequad = mantid.CropWorkspace(InputWorkspace=cloned_workspace,
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_model.py]:
```suggestion
        mantid.CropWorkspace(InputWorkspace=cloned_workspace,
```

**Review by dtasev — CHANGES_REQUESTED** [review]
Getting "OUTPUTWORKSPACE" not set error. Last of the output, including the error:

```
CropWorkspace started
Algorithm: CropWorkspace v1
Execution Date: 2019-07-12 08:05:25.037435000
Execution Duration: -1 seconds
UUID: 177c505a-4c30-4e47-b556-807300b5fa09
Parameters:
  Name: InputWorkspace, Value: __TMP0x7faaf835df60, Default?: No, Direction: Input
  Name: OutputWorkspace, Value: , Default?: Yes, Direction: Output
  Name: XMin, Value: 0.10000000000000001, Default?: No, Direction: Input
  Name: XMax, Value: 15, Default?: No, Direction: Input
  Name: StartWorkspaceIndex, Value: 0, Default?: Yes, Direction: Input
  Name: EndWorkspaceIndex, Value: 2147483647, Default?: Yes, Direction: Input
Property "OUTPUTWORKSPACE" is not set to a valid value: "Enter a name for the Output workspace".
Property "OUTPUTWORKSPACE" is not set to a valid value: "Enter a name for the Output workspace".
```

**Comment by AnthonyLim23** [comment]
could you send me instructions to try and reproduce it?

**Comment by DTasev** [comment]
I did these steps:
Open Mantidplot
Open Muon Analysis (the original one not muon analysis 2)
load MUSR 62260
Open frequency domain analysis old
Change workspace to PhaseQuad
Click Calculate FFT

On `Calculate FFT` it shows the error above. Also happens with the HIFI instructions

**Review by dtasev — APPROVED** [review]
Flake8 is happy now, functionality works well - PhaseQuad workspaces are inside the WS groups, and no errors/warnings are shown

**Pull request merged**